aea/skills/tasks.py

```diff
@@ -64,7 +63,7 @@ def __call__(self, *args: Any, **kwargs: Any) -> "Task":
         finally:
             self._is_executed = True
             self.teardown()
-        return self
```

Comment:
This breaks what we have in the gym skill. Why not return self? This let's you check task status etc.